File: content/src/main/java/org/sandboxpowered/silica/minecraft/blocks/BedBlock.java
package org.sandboxpowered.silica.minecraft.blocks;

import org.sandboxpowered.api.block.BaseBlock;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.Properties;
import org.sandboxpowered.api.state.StateFactory;
import org.sandboxpowered.api.state.property.EnumProperty;

public class BedBlock extends BaseBlock {
    private static final EnumProperty<BedPart> BED_PART = EnumProperty.of("part", BedPart.class);
    public BedBlock(Settings settings) {
        super(settings);
    }

    @Override
    public void appendProperties(StateFactory.Builder<Block, BlockState> builder) {
        super.appendProperties(builder);

        builder.add(Properties.HORIZONTAL_FACING, Properties.OCCUPIED, BED_PART);
    }

    public static enum BedPart {
        HEAD,
        FOOT
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/block/BedBlock.kt
package org.sandboxpowered.silica.block

import org.sandboxpowered.silica.block.SilicaBlockProperties.BED_PART
import org.sandboxpowered.silica.block.SilicaBlockProperties.HORIZONTAL_FACING
import org.sandboxpowered.silica.block.SilicaBlockProperties.OCCUPIED
import org.sandboxpowered.silica.state.StateProvider
import org.sandboxpowered.silica.state.block.BlockState
import org.sandboxpowered.silica.util.Identifier

class BedBlock(identifier: Identifier) : BaseBlock(identifier) {
    override fun appendProperties(builder: StateProvider.Builder<Block, BlockState>) {
        super.appendProperties(builder)
        builder.add(OCCUPIED, HORIZONTAL_FACING, BED_PART)
    }
}